packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -288,68 +237,75 @@ export default createRule<[], MessageIds>({
     }
 
     return {
-      CallExpression(node): void {
-        if (node.arguments.length === 0 || !isPromiseCatchAccess(node.callee)) {
+      CallExpression({ arguments: args, callee }): void {
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
           return;
         }
 
-        const firstArgument = node.arguments[0];
+        const staticMemberAccessKey = getStaticMemberAccessKey(
+          callee,
+          context.sourceCode.getScope(callee),
+        );
+        if (!staticMemberAccessKey) {
+          return;
+        }
 
-        // Deal with some special cases around spread element args.
-        // promise.catch(...handlers), promise.catch(...handlers, ...moreHandlers).
-        if (firstArgument.type === AST_NODE_TYPES.SpreadElement) {
-          if (node.arguments.length === 1) {
-            if (shouldFlagSingleSpreadArg(firstArgument)) {
-              context.report({
-                node: firstArgument,
-                messageId: 'useUnknown',
-              });
-            }
-          } else if (shouldFlagMultipleSpreadArgs(node.arguments)) {
-            context.report({
-              node,
-              messageId: 'useUnknownSpreadArgs',
-            });
-          }
+        const promiseMethodInfo = (
+          [
+            { method: 'catch', append: '', argIndexToCheck: 0 },
+            { method: 'then', append: ' rejection', argIndexToCheck: 1 },
+          ] satisfies {
+            method: string;
+            append: string;
+            argIndexToCheck: number;
+          }[]
```

Comment:
> make a type for this array - the trouble with inferring it is that one can omit method or append, or add extraneous properties, and the type checker won't complain.

`satisfies` is the only way to get the type checker to complain about extraneous properties — any type annotation or type parameter anywhere else and the type checker will still be OK with extraneous properties 👍